std-bits/table/src/main/java/org/enso/table/util/NameDeduplicator.java

```diff
@@ -1,35 +1,91 @@
 package org.enso.table.util;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
+
+import org.enso.base.text.UnicodeNormalizedTextEquivalence;
+import org.enso.table.data.table.Column;
 import org.enso.table.problems.Problem;
 import org.enso.table.util.problems.DuplicateNames;
 import org.enso.table.util.problems.InvalidNames;
+import org.graalvm.collections.EconomicSet;
 
 public class NameDeduplicator {
-  private final Set<String> usedNames = new HashSet<>();
+  private final EconomicSet<String> usedNames = EconomicSet.create(new UnicodeNormalizedTextEquivalence());
```

Comment:
We already are using `org.graalvm.collections` in a few places.

Mostly because, as is the case here, we need a custom `Equivalence` relation - and the Graal collections give us the ability to define that simply - with Java collections it is more complicated (we'd have to wrap in some wrappers to customize the equality, or use some third-party implementations, possibly Guava could have these, but currently we do not depend on Guava there).

Reply:
If you think that these collections are bad for performance, we can switch to some third party implementation. The docs sound like they should be decent. We are already using some Graal stuff like `org.graalvm.polyglot`, so I didn't see harm in using `org.graalvm.collections`. But if it's not good, then we can move away from them - if it's justified. Still, I'd prefer to do this as a separate ticket because the usage of these collections spans more than just the case in this PR.

Reply:
> Mostly because, as is the case here, we need a custom Equivalence relation

I see. That makes sense.

> But if it's not good, then we can move away from them - if it's justified

I don't think `org.graalvm.collections` are bad. This was more of a stylistic issue.